src/components/Root/index.tsx

```diff
@@ -79,6 +80,7 @@ const RootConsumer = (): React.ReactElement | null => {
 const Root = (): React.ReactElement => (
   <>
     <LegacyRouteRedirection history={history} />
+    <XDaiGCRedirection history={history} />
```

Comment:
This has to be further up in the tree than `AppRoutes` otherwise `switchNetworkWithUrl` tracks `xdai:0x123` before the redirection.